text/metrics/0108-naming-conventions.md

```diff
@@ -0,0 +1,55 @@
+# Metric instrument naming conventions
+
+## Purpose
+
+Names and labels for metric instruments are primarily how humans interact with metric data -- users rely on these names to build dashboards and perform analysis. The names and hierarchical structure need to be understandable and discoverable during routine exploration -- and this becomes critical during incidents.
+
+## Guidelines
+
+Metric names and labels exist within a single universe and a single hierarchy. Metric names and labels should be considered within the universe of all existing metric names. When defining new metric names and labels, consider the prior art of existing standard metrics and metrics from frameworks/libraries.
+
+Nest similar metrics together in a hierarchy based on their usage. Define a top-level hierarchy for common metric categories: for OS metrics, like CPU and network; for app runtimes, like GC internals. Libraries and frameworks should nest their metrics into a hierarchy as well. This aids in discovery and adhoc comparison. This allows a user to find similar metrics given a certain metric.
+
+The hierarchical structure of metrics defines the namespacing. Supporting OpenTelemetry artifacts define the metric structures and hierarchies for some categories of metrics, and these can assist decisions when creating future metrics.
+
+Provide consistent names for common labels. This aids in discoverability and disambiguates similar labels to metric names.
+
+"As a rule of thumb, **aggregations** over all the dimensions of a given metric should be meaningful," as Prometheus recommends.
+
+Avoid semantic ambiguity. Use prefixed metric names in cases where similar metrics have significantly different implementations across the breadth of all existing metrics. For example, every garbage collected runtime has a slightly different strategies and measures. Using common metric names for GC, not namespaced by the runtime, could create dissimilar comparisons and confusion for end users. (For example, prefer `runtime.java.gc*` over `runtime.gc*`.) Measures of many operating system metrics are similar.
+
+For conventional metrics or metrics that have their units covered by OpenTelemetry metadata (eg `Metric.WithUnit` in Go), do not include the units in the metric name. Units may be included when it provides additional meaning to the metric name. Metrics must, above all, be understandable and usable.
+
+## End of document
+
+The below text is for threaded discussion and will be deleted when ready to merge.
+
+Goal heading out of this document: when this merges, start writing specific semantic conventions for os, database, etc. Forthcoming docs will address specific metric: Network, OS, OS-agnostic
+
+
+
+## TODO TODO Working Area and Outstanding Questions TODO TODO
+
+### Fundamental questions - do we need to address this?
+* What is a metric?
+* What separates a metric from a label?
+
+### General questions
+
+~Instrumentation library != namespace~ (see note above)
```

Comment:
It is not clear which node is referred to as `namespace` is rather ambiguous here and nothing else mentions instrumentation libraries.

Reply:
Revised, thanks for pointing this out!

Reply:
This was from a discussion section of the doc and isn't part of the final doc. Resolving.